src/main/java/shapes/IsoscelesTriangle.java

```diff
@@ -0,0 +1,96 @@
+package shapes;
+
+import static java.lang.Math.*;
+
+public class IsoscelesTriangle implements Shape {
+
+    private final double sideLength;
+    private final double baseLength;
+
+    public double getSideLength() {
+        return sideLength;
+    }
+
+    public double getBaseLength() {
+        return baseLength;
+    }
+
+    public double getHeight() {
+        return sqrt(pow(sideLength, 2) - (pow(baseLength, 2) / 4.0));
+    }
+
+    @Override
+    public double getSurfaceArea() {
+        return baseLength * getHeight() / 2.0;
+    }
+
+    @Override
+    public double getCircuit() {
+        return 2 * sideLength + baseLength;
+    }
+
+    private IsoscelesTriangle(double sideLength, double baseLength) {
+        if ((2 * sideLength) <= baseLength) {
+            throw new IllegalArgumentException("2.0 times side length should be bigger than base length");
```

Comment:
Taki wariant byłby łatwiejszy do zrozumienia
```suggestion
            throw new IllegalArgumentException("side length should be bigger 2 times than base length");
```